server/queue/queue-api/src/test/java/org/apache/james/queue/api/ManageableMailQueueContract.java

```diff
@@ -0,0 +1,642 @@
+
+        long size = getManageableMailQueue().getSize();
+
+        assertThat(size).isEqualTo(1L);
+    }
+
+    @Test
+    default void processedMailsShouldNotDecreaseSize() throws Exception {
+        getManageableMailQueue().enQueue(FakeMail.builder()
+            .mimeMessage(createMimeMessage())
+            .sender(MailAddressFixture.OTHER_AT_LOCAL)
+            .recipients(MailAddressFixture.ANY_AT_JAMES, MailAddressFixture.OTHER_AT_JAMES)
+            .lastUpdated(new Date())
+            .build());
+
+        getManageableMailQueue().deQueue();
+
+        long size = getManageableMailQueue().getSize();
+
+        assertThat(size).isEqualTo(1L);
+    }
+
+    @Test
+    default void browseShouldReturnEmptyByDefault() throws Exception {
+        ManageableMailQueue.MailQueueIterator items = getManageableMailQueue().browse();
+
+        assertThat(items).isEmpty();
+    }
+
+    @Test
+    default void browseShouldReturnSingleElement() throws Exception {
+        getManageableMailQueue().enQueue(FakeMail.builder()
+            .name(NAME1)
+            .mimeMessage(createMimeMessage())
+            .sender(MailAddressFixture.OTHER_AT_LOCAL)
+            .recipients(MailAddressFixture.ANY_AT_JAMES, MailAddressFixture.OTHER_AT_JAMES)
+            .lastUpdated(new Date())
+            .build());
+
+        ManageableMailQueue.MailQueueIterator items = getManageableMailQueue().browse();
+
+        assertThat(items).extracting(ManageableMailQueue.MailQueueItemView::getMail)
+            .extracting(Mail::getName)
+            .containsExactly(NAME1);
+    }
+
+    @Test
+    default void browseShouldReturnOrderElements() throws Exception {
+        getManageableMailQueue().enQueue(FakeMail.builder()
+            .name(NAME1)
+            .mimeMessage(createMimeMessage())
+            .sender(MailAddressFixture.OTHER_AT_LOCAL)
+            .recipients(MailAddressFixture.ANY_AT_JAMES, MailAddressFixture.OTHER_AT_JAMES)
+            .lastUpdated(new Date())
+            .build());
+        getManageableMailQueue().enQueue(FakeMail.builder()
+            .name(NAME2)
+            .mimeMessage(createMimeMessage())
+            .sender(MailAddressFixture.OTHER_AT_LOCAL)
+            .recipients(MailAddressFixture.ANY_AT_JAMES, MailAddressFixture.OTHER_AT_JAMES)
+            .lastUpdated(new Date())
+            .build());
+        getManageableMailQueue().enQueue(FakeMail.builder()
+            .name(NAME3)
+            .mimeMessage(createMimeMessage())
+            .sender(MailAddressFixture.OTHER_AT_LOCAL)
+            .recipients(MailAddressFixture.ANY_AT_JAMES, MailAddressFixture.OTHER_AT_JAMES)
+            .lastUpdated(new Date())
+            .build());
+
+        ManageableMailQueue.MailQueueIterator items = getManageableMailQueue().browse();
+
+        assertThat(items).extracting(ManageableMailQueue.MailQueueItemView::getMail)
+            .extracting(Mail::getName)
+            .containsExactly(NAME1, NAME2, NAME3);
```

Comment:
FIRST, SECOND, THIRD?

Reply:
IMO without NAME1 and ci constant it is enough readable... cf antoine commits